### DISCUSSION TITLE
[Bug]: preinstalling netclient in packer images locks build image hostname into the image

### DISCUSSION THREAD

**Discussion by nepeat** [gravitl/netmaker#2229] [Q&A] [answered]
### What happened?

I am provisioning hosts with netclient preinstalled in their base image as part of a personal project that uses disposable cloud nodes. When I enroll the hosts into my netmaker network, they look like they suceed but have some odd side effects which cause them to be unable to acquire or connect to the network.

```
root@hetzner-nbg1-3:/etc/netclient# netclient join -v 4 -t XXXXX=
[netclient] 2023-04-20 20:58:56 [netclientutils_linux.go-43] IsBridgeNetwork(): Interface is a bridge network: docker0
registered with server XXXXXX.XXX
```

For each of the nodes, they have registered under the build image's hostname `packer-XYZ` but I assume problems happen as a result of multiple nodes sharing the same hostname.

I found that if I do not include Netclient in my base image and install it during the server provisioning process, connections are able to be made and IPs can be acquired for my internal mesh.

### Version

v0.18.6

### What OS are you using?

Linux

### Contributing guidelines

- [X] Yes, I did.

**Comment by nepeat** [comment]
I suspect my issues with connecting preimaged hosts is due to the fact that there is a lot of host specific information that gets stored in /etc/netclient/netclient.yaml, causing the server to think the dozen+ hosts I have provisioned and connected are the same client with the same hostname. Let me know if other details are needed or if this is able to be reproduced on your side.

Code bits of how my images are made can be found here.
1. Netclient installation point https://github.com/general-programming/megarepo/blob/mainline/automation/deploy_server/common/00-install-packages.sh#L72
2. Specific Packer image I am using https://github.com/general-programming/megarepo/blob/mainline/automation/deploy_server/hetzner/packer/app.pkr.hcl

**Comment by mattkasun** [comment]
netmaker/netclient is fine with mulitple hosts with the same hostname.  The only thing that could be causing an issue is if all the hosts have the same macaddress.

**Comment by nepeat** [comment]
All the hosts have different MAC addresses, this issue seems to be surfacing due to configuration files that are created during the base image creation process.

**Comment by mattkasun** [comment, marked as answer]
suggestion:  when deploying the image:

1. stop the netclient daemon: `systemctl stop netclient`
2. delete the /etc/netclient dir
3. start netclient daemon which will recreate the config files in /etc/netclient : `systemctl start netclient`

**Reply by nepeat** [reply]
Thanks, that worked for getting it to work as part of my images. Would it be possible to get this documented so that others creating similar setups as me will not hit this roadblock? 0.17 did not have this behaviour, so having these issues suddenly was a surprise when migrating to 0.18.

**Comment by mattkasun** [comment]
converting to discussion
